### PR TITLE
API includes sort_order, and WebSocket includes data

### PR DESCRIPTION
**Pull request by shazm**
#### Summary

These changes unblock some future feature work on the mobile v2 (gekidou).

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-40470

#### Release Note

```release-note
Adds sort order to category api, and includes category data in websocket category update event
```

### CODE REVIEW

**Comment by shazm** [comment]
> The changes look good to me. Is it reasonable to add an (api?) test to confirm the ws message contains the right payload?

I don't think that's necessary as the models have a lot of type safety already.

Rebased off the latest master, and should be good to merge.

**Comment by shazm** [comment]
> > I don't think that's necessary as the models have a lot of type safety already.
> 
> My comment was not about the correct type of the data, but instead about verify that the data is included in the first place. (Which is what this PR fixes)

Ah whoops, I understand - will see if I can add a test to ensure the keys are correct in the call

**Review by koox00 — APPROVED** [review]
LGTM!

PS: If you need an example of an API level test you can look at https://github.com/mattermost/mattermost-server/blob/master/api4/user_test.go#L5873

**Comment by Willyfrog** [comment]
@shazm do you need any help on this one? anything specific?

**Comment by enahum** [comment]
@mkraft can you help review, guide and merge this PR ASAP? the idea is that this ships as early as possible to help reduce requests on mobile v2 as soon as it ships.

Thanks.

**Review by hanzei — APPROVED** [review]
I would still like to see a test for the changes `app/channel_category.go`, but I also don't want to block the PR any further.

**Review by mkraft — APPROVED** [review]

**Comment by mkraft** [comment]
> I would still like to see a test for the changes `app/channel_category.go`, but I also don't want to block the PR any further.

@hanzei I've added a test that makes a request endpoint via the API client then ensure the WS contains the payload that @shaz has added.

**Review by hanzei — APPROVED** [review]
Awesome :tada:

Thanks @mkraft for going the extra mile and adding a test :+1:

File: api4/channel_category_test.go [hunk 1]
```diff
@@ -71,6 +73,82 @@ func TestCreateCategoryForTeamForUser(t *testing.T) {
...
+				},
+				Channels: []string{th.BasicChannel.Id},
+			},
+		}
+
+		testCategories, _, err = th.Client.UpdateSidebarCategoriesForTeamForUser(th.BasicUser.Id, th.BasicTeam.Id, testCategories)
+		require.NoError(t, err)
+
+		b, err := json.Marshal(testCategories)
+		require.Nil(t, err)
```
Comment (hanzei) [inline api4/channel_category_test.go:128]:
```suggestion
		require.NoError(t, err)
```

**Review by josephbaylon — APPROVED** [review]
LGTM. `sort_order` is now being returned in the response

```
{"categories":[{"id":"favorites_54yfk775i7gziyzkpwzwsnb9ye_jjy3iqenutn7iq793w8h6p8q3e","user_id":"54yfk775i7gziyzkpwzwsnb9ye","team_id":"jjy3iqenutn7iq793w8h6p8q3e","sort_order":0,"sorting":"","type":"favorites","display_name":"Favorites","muted":false,"collapsed":false,"channel_ids":[]},{"id":"cgogpzxuftgztpa65kiok5atty","user_id":"54yfk775i7gziyzkpwzwsnb9ye","team_id":"jjy3iqenutn7iq793w8h6p8q3e","sort_order":10,"sorting":"","type":"custom","display_name":"NEW CATEGORY","muted":false,"collapsed":false,"channel_ids":[]},{"id":"channels_54yfk775i7gziyzkpwzwsnb9ye_jjy3iqenutn7iq793w8h6p8q3e","user_id":"54yfk775i7gziyzkpwzwsnb9ye","team_id":"jjy3iqenutn7iq793w8h6p8q3e","sort_order":20,"sorting":"manual","type":"channels","display_name":"Channels","muted":false,"collapsed":false,"channel_ids":["xmikboo14prnpq5wsm3deycjqe","k5dzkn58r788jbr1gkqpa6qxye","j8ghehahr3fr3xebzdxxoi6ruc"]},{"id":"direct_messages_54yfk775i7gziyzkpwzwsnb9ye_jjy3iqenutn7iq793w8h6p8q3e","user_id":"54yfk775i7gziyzkpwzwsnb9ye","team_id":"jjy3iqenutn7iq793w8h6p8q3e","sort_order":30,"sorting":"recent","type":"direct_messages","display_name":"Direct Messages","muted":false,"collapsed":false,"channel_ids":[]}],"order":["favorites_54yfk775i7gziyzkpwzwsnb9ye_jjy3iqenutn7iq793w8h6p8q3e","cgogpzxuftgztpa65kiok5atty","channels_54yfk775i7gziyzkpwzwsnb9ye_jjy3iqenutn7iq793w8h6p8q3e","direct_messages_54yfk775i7gziyzkpwzwsnb9ye_jjy3iqenutn7iq793w8h6p8q3e"]}
```

**Pull request merged**